fossunited/api/reviewer.py

```diff
@@ -0,0 +1,311 @@
+                submission["review_status"] = "Reviewed"
+                submission["reviewer_status"] = review.to_approve
+                submission["reviewer_remarks"] = review.remarks
+                submissions_list.append(submission)
+
+    return submissions_list
+
+
+def has_reviewer_role():
+    return bool(
+        frappe.db.exists(
+            "Has Role",
+            {"role": "CFP Reviewer", "parent": frappe.session.user},
+        )
+    )
+
+
+@frappe.whitelist()
+def get_events_by_open_cfp() -> list:
+    """
+    Get all the upcoming events with open CFP
+
+    Returns:
+        list: List of events with open CFP
+    """
+    if not has_reviewer_role():
+        frappe.throw("Unauthorized Access")
+
+    cfps_to_review = []
+
+    events = frappe.db.get_list(
+        "FOSS Chapter Event",
+        filters={
+            "status": ["in", ["Approved", "Live"]],
+            "is_published": 1,
+            "event_start_date": [">=", frappe.utils.nowdate()],
+        },
+        fields=[
+            "name",
+            "event_name",
+            "event_start_date",
+            "event_end_date",
+            "chapter",
+        ],
+        page_length=20,
+        order_by="event_start_date",
+    )
+
+    for event in events:
+        cfp_exists = frappe.db.exists(
+            "FOSS Event CFP", {"event": event.name}
+        )
+        if not cfp_exists:
+            continue
+
+        cfp = frappe.db.get_value(
+            "FOSS Event CFP",
+            {"event": event.name},
+            ["name", "chapter"],
+            as_dict=1,
+        )
+        chapter = frappe.db.get_value(
+            "FOSS Chapter",
+            event.chapter,
+            ["name", "chapter_name", "chapter_type"],
+            as_dict=1,
+        )
+        submission_count = frappe.db.count(
+            "FOSS Event CFP Submission", {"linked_cfp": cfp.name}
+        )
+        cfps_to_review.append(
+            {
+                "event": event.name,
+                "event_name": event.event_name,
+                "start_date": event.event_start_date,
+                "end_date": event.event_end_date,
+                "cfp": cfp.name,
+                "submission_count": submission_count,
+                "chapter": chapter.name,
+                "chapter_name": chapter.chapter_name,
+                "chapter_type": chapter.chapter_type,
+            }
+        )
+
+    return cfps_to_review
+
+
+@frappe.whitelist()
+def has_cfp_review(
+    submission_id: str, reviewer: str = frappe.session.user
+) -> bool:
+    """
+    Check if the reviewer has reviewed the submission
+
+    Args:
+        submission_id (str): The id of the submission
+        reviewer (str): The reviewer's email
+
+    Returns:
+        bool: True if the reviewer has reviewed the submission, False otherwise
+    """
+
+    reviewer_profile = frappe.db.get_value(
+        "FOSS User Profile", {"email": reviewer}, "name"
```

Comment:
Same here as well, maybe do it for all the variables using Doctype FOSS User Profile. In my next PR which will for adding the chapter constants, I'll replace all the rest with constants.

Reply:
Will be doing the constant changes later on, in another PR. This branch is not updated to latest commit, and merging branches is making the commits dirty